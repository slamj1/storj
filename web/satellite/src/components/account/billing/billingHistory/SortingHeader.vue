// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="sort-header-container">
        <div class="sort-header-container__item date">
            <p class="sort-header-container__item__name">Date</p>
        </div>
        <div class="sort-header-container__item description">
            <p class="sort-header-container__item__name">Description</p>
        </div>
        <div class="sort-header-container__item status">
            <p class="sort-header-container__item__name">Status</p>
        </div>
        <div class="sort-header-container__item amount">
            <p class="sort-header-container__item__name">Amount</p>
        </div>
        <div class="sort-header-container__item download">
            <p class="sort-header-container__item__name">Download</p>
        </div>
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

@Component
export default class SortingHeader extends Vue {}
</script>

<style scoped lang="scss">
    .sort-header-container {
        display: flex;
        background-color: rgba(245, 246, 250, 0.5);
        width: calc(100% - 60px);
        padding: 0 30px;

        &__item {
            text-align: left;
            min-width: 25%;

            &__name {
                font-family: 'font_medium', sans-serif;
                font-size: 16px;
                line-height: 10px;
                color: #afb7c1;
            }
        }
    }

    .description {
        min-width: 31%;
    }

    .status {
        min-width: 12%;
    }

    .amount {
        min-width: 22%;
        margin: 0;
    }

    .download {
        text-align: right;
        margin: 0;
        min-width: 10%;
    }
</style>
