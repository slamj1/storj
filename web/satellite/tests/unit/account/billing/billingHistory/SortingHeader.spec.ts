// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import SortingHeader from '@/components/account/billing/billingHistory/SortingHeader.vue';

import { createLocalVue, mount } from '@vue/test-utils';

const localVue = createLocalVue();

describe('SortingHeader', (): void => {
    it('renders correctly', (): void => {
        const wrapper = mount(SortingHeader, {
            localVue,
        });

        expect(wrapper).toMatchSnapshot();
    });
});
