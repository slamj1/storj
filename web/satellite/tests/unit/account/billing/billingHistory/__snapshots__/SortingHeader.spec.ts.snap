// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortingHeader renders correctly 1`] = `
<div class="sort-header-container">
  <div class="sort-header-container__item date">
    <p class="sort-header-container__item__name">Date</p>
  </div>
  <div class="sort-header-container__item description">
    <p class="sort-header-container__item__name">Description</p>
  </div>
  <div class="sort-header-container__item status">
    <p class="sort-header-container__item__name">Status</p>
  </div>
  <div class="sort-header-container__item amount">
    <p class="sort-header-container__item__name">Amount</p>
  </div>
  <div class="sort-header-container__item download">
    <p class="sort-header-container__item__name">Download</p>
  </div>
</div>
`;
